api/envoy/config/wasm/v2/wasm.proto

```diff
@@ -0,0 +1,68 @@
+syntax = "proto3";
+
+package envoy.config.wasm.v2;
+
+option java_outer_classname = "WasmProto";
+option java_multiple_files = true;
+option java_package = "io.envoyproxy.envoy.config.wasm.v2";
+
+import "envoy/api/v2/core/base.proto";
+
+import "validate/validate.proto";
+
+// [#protodoc-title: Wasm service]
+
+// Configuration for a Wasm VM.
+// [#next-free-field: 6]
+message VmConfig {
+  // An ID which will be used along with a hash of the wasm code (or null_vm_id) to determine which
+  // VM will be used for the plugin. All plugins which use the same vm_id and code will use the same
+  // VM. May be left blank.
```

Comment:
Can you elaborate on why you would want to share a VM? How would an end user make this call? Please put the answer in comment form :)

Reply:
https://docs.google.com/document/d/1YPbxuEIPwKsW4Ft73diHRBTUNbYIlrQHNnS9dXdn778/edit?usp=sharing

TL;DR when you have 20 plugins and 32 cores you get  ~1280 VMs, each of which requires its own stack and heap which could end up being a lot of memory.   Also because v8 uses guard pages to avoid bound checks each VM uses 4GB of virtual memory which can exhaust virtual memory.  Also combining plugins into the same VM allows them to share state cheaply and easily.

Is this the kind of comment you are looking for or is it TMI?

Reply:
I think in the comment, you should point out that there is a combinatorial explosion in those parameters, which is why you end up reusing VMs. I think the reader needs to know when they should reuse (should they always do this by defaults? Should they only do it if they find themselves running out of memory? Should they ensure that VMs aren't reused across distinct security contexts?).

Reply:
That is a lot of information to include in a comment.  Perhaps a TL;DR in the comment and a reference to a more substantial discussion in a doc/reference material.

Reply:
Sounds good, please link to it with a `:ref:`

Reply:
I have an additional concern here. We're pushing a lot of proxy-specific VM management out to the control plane. Will this scale? How much proxy state needs to be tracked and managed in this way? How do we envisage VM IDs being used in practice?

Reply:
Moving this to the architecture doc.